File: JavaScript/Array & String/1.3.js
#!/usr/bin/env node

/************************************
              cc150
    1.3 judge anagram string
    Assume the string is ASCII
    No space or capital

************************************/


function permutation(string1, string2) {

    var string1 = string1.replace(/\s+/g, '').toLowerCase();
    var string2 = string2.replace(/\s+/g, '').toLowerCase();

    if (string1.length !== string2.length) {
        return false
    }

    var charSet = Array.apply(null, new Array(128))
                  .map(Number.prototype.valueOf, 0);

    for (i = 0, j = string1.length; i < j; i++) {
        var charVal = string1.charCodeAt(i);
        ++charSet[charVal];
    }

    for (i = 0, j = string2.length; i < j; i++) {
        var charVal = string2.charCodeAt(i);
        if (--charSet[charVal] < 0) {
            return false;
        }
    }

    return true;
}

var args = process.argv.slice(0, 4);

console.log(permutation(args[2], args[3]));

File: Python/Array & String/1.3.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

#************************************
#              cc150
#  1.3 judge anagram string
#  Assume the string is ASCII
#  No space or capital
#
#************************************

import sys


def permutation(string1, string2):

	string1 = string1.replace(" ", "").lower()
	string2 = string2.replace(" ", "").lower()

	if len(string1) != len(string2):
		return False

	char_set = [0] * 128

	for char in string1:
		index = ord(char)
		char_set[index] += 1

	for char in string2:
		index = ord(char)
		char_set[index] -= 1
		if char_set[index] < 0:
			return False

	return True


if __name__ == '__main__':

	if len(sys.argv) != 3:
		print('Usage: python3 1.1.py string1 string2.')
	else:
		print(permutation(sys.argv[1], sys.argv[2]))